tbot_contrib/locking.py

```diff
@@ -186,6 +187,7 @@ def _lock_try_acquire(self, name: str, expiry: typing.Optional[int]) -> bool:
         else:  # else store the Unix time timestamp of expiry
             time_str = str(int(self.exec0("date", "+%s")) + expiry)
         shell_pid = self.env("$") if self.lock_checkpid else "0"
+        self.exec0("chmod", "0666", tempfile)
         tempfile.write_text(f"{time_str} {shell_pid}\n")
         # Try linking lockfile, hard-linking is atomic
         lock_acquired = self.test("ln", tempfile, lockfile)
```

Comment:
Just for my personal reference, the `ln` hardlinking operation copies permissions exactly, so we do not need to also set permissions for the second linked name.